src/neo/Network/P2P/Payloads/StateRoot.cs

```diff
@@ -0,0 +1,122 @@
+using Neo.Cryptography;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using System;
+using System.IO;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class StateRoot : ICloneable<StateRoot>, IInventory
```

Comment:
If we decide to save it in block header, i think that this wont be required.

Reply:
Yes. But Erik doesn't agree.